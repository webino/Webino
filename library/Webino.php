<?php
/**
 * Webino™ (http://webino.sk)
 *
 * @link        https://github.com/webino for the canonical source repository
 * @copyright   Copyright (c) 2015-2018 Webino, s.r.o. (http://webino.sk)
 * @author      Peter Bačinský <peter@bacinsky.sk>
 * @license     BSD-3-Clause
 */

use Webino\Version;
use Webino\App\Application\CoreConfig;
use Webino\App\Factory;
use Webino\App\Options\DebuggerOptions;
use Webino\App\Service\Debugger;
use Webino\App\Service\DebuggerInterface;

/**
 * Class Webino
 */
class Webino
{
    /**
     * Webino™ version
     */
    const VERSION = Version::VERSION;

    /**
     * Create application core config
     *
     * @param array $config
     * @return CoreConfig
     */
    public static function config(array $config)
    {
        return new CoreConfig($config);
    }

    /**
     * Create Webino application
     *
     * @param array|object $config
     * @param DebuggerInterface $debugger
     * @return \WebinoAppLib\Application\AbstractBaseApplication
     */
    public static function application($config = null, DebuggerInterface $debugger = null)
    {
        return (new Factory)->create($config, $debugger);
    }

    /**
     * Create application debugger
     *
     * @param array|DebuggerOptions $options
     * @return Debugger
     */
    public static function debugger($options = [])
    {
        return new Debugger($options);
    }

    /**
     * Create application debugger options
     *
     * @return DebuggerOptions
     */
    public static function debuggerOptions()
    {
        return new DebuggerOptions;
    }
}
