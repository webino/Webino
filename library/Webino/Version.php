<?php
/**
 * Webino™ (http://webino.sk)
 *
 * @link        https://github.com/webino for the canonical source repository
 * @copyright   Copyright (c) 2015-2017 Webino, s.r.o. (http://webino.sk)
 * @author      Peter Bačinský <peter@bacinsky.sk>
 * @license     BSD-3-Clause
 */

namespace Webino;

/**
 * Class to store and retrieve the version of Webino™
 */
final class Version
{
    /**
     * Webino™ version identification
     */
    const VERSION = '0.1.0';
}
