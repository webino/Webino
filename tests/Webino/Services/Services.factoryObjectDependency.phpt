<?php
/**
 * Webino™ (http://webino.sk)
 *
 * @link        https://github.com/webino for the canonical source repository
 * @copyright   Copyright (c) 2015-2018 Webino, s.r.o. (http://webino.sk)
 * @author      Peter Bačinský <peter@bacinsky.sk>
 * @license     BSD-3-Clause
 */

use Tester\Assert;
use Webino\Services\Factories\AbstractFactory;
use Webino\Services\ServiceContainer;
use Webino\Services\ServiceContainerInterface;

require __DIR__ . '/../../bootstrap.php';


class TestService
{

}

class TestServiceFactory extends AbstractFactory
{
    public $service;

    public function __construct($service)
    {
        $this->service = $service;
    }

    public function createService(ServiceContainerInterface $services)
    {
        return $this->service;
    }
}


$services = new ServiceContainer;
$service = new TestService;


$services->set(TestServiceFactory::class, function (ServiceContainerInterface $services) use ($service) {
    return new TestServiceFactory($service);
});

$services->set(TestService::class, TestServiceFactory::class);


Assert::true($services->has(TestService::class));
Assert::same($service, $services->get(TestService::class));
