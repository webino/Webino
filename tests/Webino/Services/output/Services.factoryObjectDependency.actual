TestServiceFactory::__set_state(/* #284f */ array(
	'service' => TestService::__set_state(/* #fce1 */ array()),
	'callback' => NULL,
))
