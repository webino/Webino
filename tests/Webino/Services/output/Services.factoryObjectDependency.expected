TestService::__set_state(/* #fce1 */ array())
